ports/mimxrt/machine_pin.c

```diff
@@ -0,0 +1,325 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Philipp Ebensberger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <stdint.h>
+
+#include "fsl_gpio.h"
+#include "fsl_iomuxc.h"
+
+#include "py/runtime.h"
+#include "py/mphal.h"
+#include "pin.h"
+#include "mphalport.h"
+
+// Local functions
+STATIC mp_obj_t machine_pin_obj_init_helper(const machine_pin_obj_t *self, size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args);
+STATIC void named_pin_obj_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind);
+
+// Class Methods
+STATIC void machine_pin_obj_print(const mp_print_t *print, mp_obj_t o, mp_print_kind_t kind);
+STATIC mp_obj_t machine_pin_obj_call(mp_obj_t self_in, mp_uint_t n_args, mp_uint_t n_kw, const mp_obj_t *args);
+STATIC mp_obj_t machine_pin_obj_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args);
+
+// Instance Methods
+STATIC mp_obj_t machine_pin_off(mp_obj_t self_in);
+STATIC mp_obj_t machine_pin_on(mp_obj_t self_in);
+STATIC mp_obj_t machine_pin_value(size_t n_args, const mp_obj_t *args);
+STATIC mp_obj_t machine_pin_mode(mp_obj_t self_in);
+STATIC mp_obj_t machine_pin_init(size_t n_args, const mp_obj_t *args, mp_map_t *kw_args);
+STATIC mp_obj_t machine_pin_drive(size_t n_args, const mp_obj_t *args);
+
+// Local data
+enum {
+    PIN_INIT_ARG_MODE = 0,
+    PIN_INIT_ARG_PULL,
+    PIN_INIT_ARG_VALUE,
+    PIN_INIT_ARG_DRIVE,
+};
+
+// Pin mapping dictionaries
+const mp_obj_type_t machine_pin_cpu_pins_obj_type = {
+    { &mp_type_type },
+    .name = MP_QSTR_cpu,
+    .print = named_pin_obj_print,
```

Comment:
this `.print` entry is not needed (I don't think it's even possible to get to this code because there are no instances of this type)

Reply:
removed